### PR TITLE
graplang: init at 1.46

### PR DESCRIPTION
**Pull request by afh**
This is an implementation of Grap the language for describing graphs as first implemented by Brian Kernighan and Jon Bentley at Bell Labs.

The output of grap can further be processed with pic and groff or TeX.

## Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [ ] x86_64-darwin
  - [x] aarch64-darwin
- For non-Linux: Is sandboxing enabled in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
  - [ ] `sandbox = relaxed`
  - [ ] `sandbox = true`
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [ ] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [ ] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [24.05 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2405.section.md) (or backporting [23.05](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md) and [23.11](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) Release notes)
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [ ] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

---

Add a :+1: [reaction] to [pull requests you find important].

[reaction]: https://github.blog/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/
[pull requests you find important]: https://github.com/NixOS/nixpkgs/pulls?q=is%3Aopen+sort%3Areactions-%2B1-desc

### CODE REVIEW

File: pkgs/by-name/gr/graplang/package.nix [hunk 1]
```diff
@@ -0,0 +1,29 @@
+{ lib, stdenv, fetchurl
+, flex, bison }:
```
Comment (AndersonTorres) [inline pkgs/by-name/gr/graplang/package.nix:2]:
```suggestion
{ lib
, stdenv
, fetchurl
, flex
, bison 
}:
```

File: pkgs/by-name/gr/graplang/package.nix [hunk 2]
```diff
@@ -0,0 +1,29 @@
...
+
+stdenv.mkDerivation {
```
Comment (AndersonTorres) [inline pkgs/by-name/gr/graplang/package.nix:4]:
```suggestion
stdenv.mkDerivation (finalAttrs: {
```

File: pkgs/by-name/gr/graplang/package.nix [hunk 3]
```diff
@@ -0,0 +1,29 @@
...
+  pname = "graplang";
+  version = "1.46";
+
+  src = fetchurl {
+    url = "https://www.lunabase.org/~faber/Vault/software/grap/grap-1.46.tar.gz";
```
Comment (AndersonTorres) [inline pkgs/by-name/gr/graplang/package.nix:9]:
```suggestion
    url = "https://www.lunabase.org/~faber/Vault/software/grap/grap-${finalAttrs.version}.tar.gz";
```

File: pkgs/by-name/gr/graplang/package.nix [hunk 4]
```diff
@@ -0,0 +1,29 @@
...
+  meta = with lib; {
+    description = "Language for typesetting graphs";
+    longDescription = ''
+      Grap is an Expressive language for describing graphs and incorporating
+      them in typeset documents.  It is implemented as a preprocessor to
+      Kernigan's pic language for describing languages, so any system that can
+      use pic can use grap. For sure, TeX and groff can use it.
+    '';
+    homepage = "https://www.lunabase.org/~faber/Vault/software/grap/";
+    changelog = "https://github.com/snorerot13/grap/blob/master/CHANGES";
```
Comment (AndersonTorres) [inline pkgs/by-name/gr/graplang/package.nix:24]: Why not to use fetchFromGitHub?

File: pkgs/by-name/gr/graplang/package.nix [hunk 5]
```diff
@@ -0,0 +1,29 @@
...
+    license = licenses.bsd2;
+    maintainers = with maintainers; [ afh ];
+    mainProgram = "grap";
+  };
+}
```
Comment (AndersonTorres) [inline pkgs/by-name/gr/graplang/package.nix:29]:
```suggestion
})
```

**Review by AndersonTorres — CHANGES_REQUESTED** [review]

File: pkgs/by-name/gr/graplang/package.nix (hunk 4 above)
Comment (afh) [reply pkgs/by-name/gr/graplang/package.nix:24]: The GitHub repo has neither [releases](https://github.com/snorerot13/grap/releases) nor [tags](https://github.com/snorerot13/grap/tags) opposed to the [homepage](https://www.lunabase.org/~faber/Vault/software/grap/) which provides a link to the versioned sources.

File: pkgs/by-name/gr/graplang/package.nix (hunk 3 above)
Comment (afh) [reply pkgs/by-name/gr/graplang/package.nix:9]: 👍 🤦

**Comment by afh** [comment]
Thanks for the review and feedback, @AndersonTorres, very helpful and much appreciated.

Do you find `graplang` to be a suitable name for this package given that another package is named [`grap`](https://github.com/NixOS/nixpkgs/blob/master/pkgs/tools/security/grap/default.nix)?

**Comment by AndersonTorres** [comment]
> Do you find graplang to be a suitable name for this package given that another package is named grap?

Yep!

**Comment by afh** [comment]
Friendly ping on this, @AndersonTorres. What is missing or needed in order to get this merged?

**Comment by AndersonTorres** [comment]
> Friendly ping on this, @AndersonTorres. What is missing or needed in order to get this merged?

Someone with merger rights :/

**Comment by afh** [comment]
@AndersonTorres I'd like to post this PR on the [PRs already reviewed channel on discourse](https://discourse.nixos.org/t/prs-already-reviewed/2617), before doing so I kindly request your formal approval on this PR.

**Review by AndersonTorres — APPROVED** [review]
Waiting mergers (I will post this on Discourse btw).

File: pkgs/by-name/gr/graplang/package.nix [hunk 6]
```diff
@@ -0,0 +1,33 @@
+{ lib
+, stdenv
+, fetchurl
+, flex
+, bison
+}:
+
+stdenv.mkDerivation (finalAttrs: {
+  pname = "graplang";
+  version = "1.46";
```
Comment (pbsds) [inline pkgs/by-name/gr/graplang/package.nix:10]:
```suggestion
  version = "1.46";
  
  outputs = [ "out" "man" ];
```

**Review by pbsds — APPROVED** [review]
Result of `nixpkgs-review pr 272864` run on x86_64-linux [1](https://github.com/Mic92/nixpkgs-review)
<details>
  <summary>1 package built:</summary>
  <ul>
    <li>graplang</li>
  </ul>
</details>

Runs fine and LGTM, just a nit

File: pkgs/by-name/gr/graplang/package.nix (hunk 6 above)
Comment (afh) [reply pkgs/by-name/gr/graplang/package.nix:10]: Thanks for taking the time to review this PR and suggesting an improvement, @pbsds, much appreciated.

**Comment by pbsds** [comment]
please squash, and fix editorconfig error

**Comment by afh** [comment]
☑️ editorconfig fixed and squashed as requested.

**Comment by afh** [comment]
Thanks for merging this, @pbsds, much appreciated 🙂

**Pull request merged**